oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/RunningRule.java

```diff
@@ -115,6 +121,24 @@ public void in(MetaInAlarm meta, Metrics metrics) {
             }
         }
 
+        if (StringUtil.isNotEmpty(includeNamesRegex)) {
+            if (!meta.getName().matches(includeNamesRegex)) {
+                if (log.isTraceEnabled()) {
+                    log.trace("{} doesn't match the include regex {}", meta.getName(), includeNamesRegex);
+                }
+                return;
+            }
+        }
+
+        if (StringUtil.isNotEmpty(excludeNamesRegex)) {
+            if (meta.getName().matches(excludeNamesRegex)) {
```

Comment:
This string#matches includes `Pattern p = Pattern.compile(regex)`, due to we have the regex expression before all checked, I think we should compile them first when loading the config or config changed.

Could you run a benchmark to show the performance difference between this and the pre-compiled match?

Reply:
I do a little benchmark, using regex `Service\_\d+` to match` Service_1`, make a loop with 10 million times. 

String#matches: 7198 millisecond
Pattern#matcher()#matches(): 977 millisecond

I will change to using the pre-build mode, thanks.